src/NumericalAlgorithms/Spectral/SwshCollocation.cpp

```diff
@@ -84,28 +85,16 @@ dispatch_to_precomputed_static_collocation_impl(
              "construct the Collocation manually, or consider (with caution) "
              "increasing `collocation_maximum_l_max`.");
   }
-  const Collocation<Representation>* collocation = nullptr;
-  auto get_collocation_if_match = [&collocation, &index ](auto i) noexcept {
-    if (UNLIKELY(decltype(i)::value == index)) {
-      static const Collocation<Representation> precomputed_collocation =
-          Collocation<Representation>{decltype(i)::value};
-      collocation = &precomputed_collocation;
-    }
-    return 0;
-  };
-  expand_pack(
-      get_collocation_if_match(std::integral_constant<size_t, Is>{})...);
-  // clang-tidy warns about null reference return, but will only return null in
-  // the erroring execution path.
-  return *collocation;  // NOLINT
+  std::array<const Collocation<Representation>& (*)(), sizeof...(Is)> cache{
```

Comment:
I'm pretty sure this would be better as `static const`; I don't think there should be any reason to want to generate the array more than once.